From bd5845569980ca496b0169ada40d289ad667fd0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B8ren=20Sandmann=20Pedersen?= <ssp@redhat.com>
Date: Tue, 15 Jan 2013 10:08:04 -0500
Subject: [PATCH 1/2] Add new DebugRenderFallbacks option

This option defaults to off. When enabled, uxa_set_fallback_debug() is
set to true. Later on more debug information may be turned on
conditional on this option.
---
 src/qxl.h        |    2 ++
 src/qxl_driver.c |   17 +++++++----------
 2 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/src/qxl.h b/src/qxl.h
index 8e2802a..b73f5b4 100644
--- a/src/qxl.h
+++ b/src/qxl.h
@@ -103,6 +103,7 @@ enum {
     OPTION_ENABLE_IMAGE_CACHE = 0,
     OPTION_ENABLE_FALLBACK_CACHE,
     OPTION_ENABLE_SURFACES,
+    OPTION_DEBUG_RENDER_FALLBACKS,
     OPTION_NUM_HEADS,
 #ifdef XSPICE
     OPTION_SPICE_PORT,
@@ -236,6 +237,7 @@ struct _qxl_screen_t
     int				enable_image_cache;
     int				enable_fallback_cache;
     int				enable_surfaces;
+    int				debug_render_fallbacks;
     
 #ifdef XSPICE
     /* XSpice specific */
diff --git a/src/qxl_driver.c b/src/qxl_driver.c
index e2d08c7..b0e755e 100644
--- a/src/qxl_driver.c
+++ b/src/qxl_driver.c
@@ -77,6 +77,8 @@ const OptionInfoRec DefaultOptions[] =
       "EnableFallbackCache",      OPTV_BOOLEAN, { 0 }, TRUE },
     { OPTION_ENABLE_SURFACES,
       "EnableSurfaces",           OPTV_BOOLEAN, { 0 }, TRUE },
+    { OPTION_DEBUG_RENDER_FALLBACKS,
+      "DebugRenderFallbacks",	  OPTV_BOOLEAN, { 0 }, TRUE },
     { OPTION_NUM_HEADS,
       "NumHeads",                 OPTV_INTEGER, { 4 }, FALSE },
 #ifdef XSPICE
@@ -1673,7 +1675,7 @@ static Bool
 setup_uxa (qxl_screen_t *qxl, ScreenPtr screen)
 {
     ScrnInfoPtr scrn = xf86ScreenToScrn (screen);
-    
+
 #if HAS_DIXREGISTERPRIVATEKEY
     if (!dixRegisterPrivateKey (&uxa_pixmap_index, PRIVATE_PIXMAP, 0))
 	return FALSE;
@@ -1706,14 +1708,7 @@ setup_uxa (qxl_screen_t *qxl, ScreenPtr screen)
 	return FALSE;
     }
     
-#if 0
-    uxa_set_fallback_debug (screen, FALSE);
-#endif
-    
-#if 0
-    if (!uxa_driver_init (screen, qxl->uxa))
-	return FALSE;
-#endif
+    uxa_set_fallback_debug (screen, qxl->debug_render_fallbacks);
     
     return TRUE;
 }
@@ -2440,7 +2435,9 @@ qxl_pre_init (ScrnInfoPtr pScrn, int flags)
     qxl->enable_fallback_cache =
         xf86ReturnOptValBool (qxl->options, OPTION_ENABLE_FALLBACK_CACHE, TRUE);
     qxl->enable_surfaces =
-        xf86ReturnOptValBool (qxl->options, OPTION_ENABLE_SURFACES, TRUE);
+	xf86ReturnOptValBool (qxl->options, OPTION_ENABLE_SURFACES, TRUE);
+    qxl->debug_render_fallbacks =
+	xf86ReturnOptValBool (qxl->options, OPTION_DEBUG_RENDER_FALLBACKS, FALSE);
     qxl->num_heads =
         get_int_option (qxl->options, OPTION_NUM_HEADS, "QXL_NUM_HEADS");
     
-- 
1.7.4

